substrate/client/rpc/src/state/state_full.rs

```diff
@@ -217,6 +216,35 @@ where
 			.map_err(client_err)
 	}
 
+	fn storage_diff(
+		&self,
+		start : Block::Hash,
+		end : Block::Hash,
+	) -> std::result::Result<Vec<(StorageKey, Option<StorageData>)>, JsonRpseeError> {
+		let mut start_keys = self.client.storage_keys(start, None, None).map_err(client_err)?;
+		let end_keys = self.client.storage_keys(end, None, None).map_err(client_err)?;
+		
+		// prepare diff
+		let mut storage_diff : Vec<(StorageKey, Option<StorageData>)> = vec![];
+		for key in end_keys {
+			// if a new key is present, add it to the diff
+			if !start_keys.contains(&key) {
+				let new_storage = self.client.storage(end, &key).map_err(client_err)?;
+				storage_diff.push((key, new_storage))
+			} else {
+				let start_storage_val = self.client.storage(start, &key).map_err(client_err)?;
+				let end_storage_val = self.client.storage(end, &key).map_err(client_err)?;
+
+				if start_storage_val != end_storage_val {
+					storage_diff.push((key, end_storage_val))
+				}
+			}
+		}
```

Comment:
This algo ignores keys that existed in `start` and then were deleted.
Such keys must be added to the result: `storage_diff.push((deleted_key, Null))`

Reply:
no, it won't
You iterate by `end_keys`: `for key in end_keys {`.
If key was deleted - it won't be included in `end_keys`

Reply:
<img width="549" alt="Screenshot 2023-10-12 at 12 55 27" src="https://github.com/traittech/polkadot-sdk/assets/2777812/8591cca2-f974-4975-9ab3-3982b84cb75e">

marked with yellow the missed keys